src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.NativeAOT.targets

```diff
@@ -110,9 +115,96 @@ This file contains the NativeAOT-specific MSBuild logic for .NET for Android.
       <ResolvedFileToPublish Include="$(_NdkSysrootDir)libc++_shared.so" RuntimeIdentifier="$(RuntimeIdentifier)" />
       <!-- Satellite assemblies -->
       <IlcSatelliteAssembly Include="$(_OuterIntermediateSatelliteAssembliesWithTargetPath)" />
+
+      <LinkerArg Include="@(RuntimePackAsset->WithMetadataValue('Filename', 'libnaot-android.$(Configuration.ToLower())-static-$(Configuration.ToLower())'))" />
+      <LinkerArg Include="$(_NdkSysrootDir)libc++_shared.so" />
+
+      <!-- Every p/invoke using the `xa-internal-api` "library" will be called directly -->
+      <DirectPInvoke Include="xa-internal-api" />
+    </ItemGroup>
+  </Target>
+
+  <!-- Make sure we have a chance to generate our application-specific static library before linking
+       takes place -->
+  <PropertyGroup>
+    <!-- This should work, but unfortunately there's a bug in the current NAOT targets which
+         overwrites `LinkNativeDependsOn` property value -->
+    <LinkNativeDependsOn>
+      _GenerateNativeAotAndroidAppLibrary;
+      $(LinkNativeDependsOn)
+    </LinkNativeDependsOn>
+  </PropertyGroup>
```

Comment:
I would maybe just remove these lines, because they don't do anything.

Can we move the comment to a place that does `BeforeTargets="LinkNative"`?